build/cmake/lib/CMakeLists.txt

```diff
@@ -42,7 +42,10 @@ if (MSVC)
     file(GLOB DecompressSources ${LIBRARY_DIR}/decompress/*.c)
     add_compile_options(-DZSTD_DISABLE_ASM)
 else ()
-    file(GLOB DecompressSources ${LIBRARY_DIR}/decompress/*.c ${LIBRARY_DIR}/decompress/*.S)
+    file(GLOB DecompressSources ${LIBRARY_DIR}/decompress/*.c)
+    if (CMAKE_SYSTEM_PROCESSOR STREQUAL "x86_64" AND NOT CMAKE_SYSTEM_NAME STREQUAL "SunOS")
+        file(GLOB DecompressASMSources ${LIBRARY_DIR}/decompress/*.S)
+    endif ()
```

Comment:
Should there be some kind of :
```
else ()
add_compile_options(-DZSTD_DISABLE_ASM)
```
when the `*.S` assembly files are not part of the source list ?